tests/ui/precedence.stderr

```diff
@@ -43,5 +43,29 @@ error: operator precedence can trip the unwary
 LL |     3 & 5 - 2;
    |     ^^^^^^^^^ help: consider parenthesizing your expression: `3 & (5 - 2)`
 
-error: aborting due to 7 previous errors
+error: operator precedence can trip the unwary
+  --> tests/ui/precedence.rs:23:5
+   |
+LL |     12 & 0xF000 << 4;
+   |     ^^^^^^^^^^^^^^^^ help: consider parenthesizing your expression: `12 & (0xF000 << 4)`
+
+error: operator precedence can trip the unwary
+  --> tests/ui/precedence.rs:24:5
+   |
+LL |     12 & 0xF000 >> 4;
```

Comment:
I think it would be clearer if, in your tests, you used examples giving different results if the wrong interpretation was made. Here, `(12 & 0xF000) >> 4` and `12 & (0xF000 >> 4)` would both give the same result (0).

For example, you could use `0x0F00 & 0xF000 >> 4`, which would give 0 with the wrong parentheses, and `0xF00` with the right ones or without parentheses.

Reply:
Thanks for your kind comment!
I have modified the tests and now if we place wrong parentheses we will get different results.